bindings/java/src/org/sleuthkit/datamodel/blackboardutils/CommunicationArtifactsHelper.java

```diff
@@ -0,0 +1,719 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.lang3.StringUtils;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardArtifact.ARTIFACT_TYPE;
+import org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE;
+import org.sleuthkit.datamodel.Account;
+import org.sleuthkit.datamodel.AccountFileInstance;
+import org.sleuthkit.datamodel.Blackboard;
+import org.sleuthkit.datamodel.BlackboardArtifact;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.DataSource;
+import org.sleuthkit.datamodel.Relationship;
+import org.sleuthkit.datamodel.SleuthkitCase;
+import org.sleuthkit.datamodel.TskCoreException;
+import org.sleuthkit.datamodel.TskDataException;
+
+/**
+ * A helper class to support modules that parse SQLite databases from mobile
+ * apps and create artifacts.
+ */
+public final class CommunicationArtifactsHelper extends ArtifactHelper {
+
+	private static final Logger logger = Logger.getLogger(CommunicationArtifactsHelper.class.getName());
+
+	/**
+	 * Enum for message read status
+	 */
+	public enum MessageReadStatus {
+
+		UNKNOWN, /// read status is unknown
+		UNREAD, /// message has not been read
+		READ        /// message has been read
+	}
+
+	/**
+	 * Enum for call/message direction
+	 */
+	public enum CommunicationDirection {
+		UNKNOWN("Unknown"),
+		INCOMING("Incoming"),
+		OUTGOING("Outgoing");
+
+		private final String dirStr;
+
+		CommunicationDirection(String dir) {
+			this.dirStr = dir;
+		}
+
+		public String getString() {
+			return dirStr;
+		}
+	}
+
+	/**
+	 * Enum for call media type
+	 */
+	public enum CallMediaType {
+		UNKNOWN("Unknown"),
+		AUDIO("Audio"),
+		VIDEO("Video");
+
+		private final String typeStr;
+
+		CallMediaType(String type) {
+			this.typeStr = type;
+		}
+
+		public String getString() {
+			return typeStr;
+		}
+	}
+
+	// 'self' account for the application. 
+	private final AccountFileInstance selfAccountInstance;
+
+	// type of accounts to be created for the Application using this helper
+	private final Account.Type accountsType;
+
+	/**
+	 * Constructs a AppDB parser helper for the given DB file.
+	 *
+	 * This is a constructor for Apps that that do not have any app specific
+	 * account information for device owner and will use a 'Device' account in
+	 * lieu.
+	 *
+	 * It creates a DeviceAccount instance to use as a self account.
+	 *
+	 * @param caseDb       Sleuthkit case db
+	 * @param moduleName   name module using the helper
+	 * @param srcFile      source file being processed by the module
+	 * @param accountsType account types created by this module
+	 *
+	 * @throws TskCoreException
+	 */
+	public CommunicationArtifactsHelper(SleuthkitCase caseDb, String moduleName, AbstractFile srcFile, Account.Type accountsType) throws TskCoreException {
```

Comment:
The documentation incorrectly refers to AppDbParser and to "apps." It is now more general.

Since this is public API documentation, it would be useful to beef it up to help module writers. What is a self account? What does it mean for the helper to create account types? We have not said a thing about this class creating both accounts and artifacts, at least until we proceed to the next constructor.